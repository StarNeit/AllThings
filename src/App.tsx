import React from 'react'
import RootLayoutContainer from 'pages/RootLayoutContainer'

class App extends React.Component {
  render() {
    return <RootLayoutContainer />
  }
}

export default App
