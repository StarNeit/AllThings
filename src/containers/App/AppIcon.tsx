import AndroidClipboardIcon from '@allthings/react-ionicons/lib/AndroidClipboardIcon'
import AndroidHomeIcon from '@allthings/react-ionicons/lib/AndroidHomeIcon'
import AndroidOpenIcon from '@allthings/react-ionicons/lib/AndroidOpenIcon'
import AndroidSettingsIcon from '@allthings/react-ionicons/lib/AndroidSettingsIcon'
import AndroidShareAltIcon from '@allthings/react-ionicons/lib/AndroidShareAltIcon'
import AndroidShareIcon from '@allthings/react-ionicons/lib/AndroidShareIcon'
import BagIcon from '@allthings/react-ionicons/lib/BagIcon'
import ClipboardIcon from '@allthings/react-ionicons/lib/ClipboardIcon'
import FlashIcon from '@allthings/react-ionicons/lib/FlashIcon'
import IosPaperOutlineIcon from '@allthings/react-ionicons/lib/IosPaperOutlineIcon'
import PeopleOutlineIcon from '@allthings/react-ionicons/lib/IosPeopleOutlineIcon'
import WandIcon from '@allthings/react-ionicons/lib/WandIcon'
// tslint:disable-next-line:no-implicit-dependencies
import { Icon, IProps as IconProps } from 'components/Icon'
import React from 'react'
import { MicroApps } from 'enums'

const iconMap = {
  archilogic: AndroidOpenIcon,
  clipboard: BagIcon,
  'community-articles': AndroidClipboardIcon,
  consumption: FlashIcon,
  documents: AndroidHomeIcon,
  'e-concierge': WandIcon,
  'external-content': AndroidOpenIcon,
  'external-link': AndroidShareIcon,
  helpdesk: ClipboardIcon,
  marketplace: BagIcon,
  'my-contracts': AndroidHomeIcon,
  project: IosPaperOutlineIcon,
  settings: AndroidSettingsIcon,
  sharing: AndroidShareAltIcon,
  unit: AndroidHomeIcon,
  'who-is-who': PeopleOutlineIcon,
}

interface IProps extends IconProps {
  readonly children?: any
  readonly microApp?: MicroApps
  readonly style: object
}

function AppIcon(props: IProps): JSX.Element {
  const { microApp, ...iconProps } = props
  const AppIconMapped = iconMap[microApp]

  return (
    <Icon {...iconProps}>
      <AppIconMapped />
    </Icon>
  )
}

export default AppIcon
