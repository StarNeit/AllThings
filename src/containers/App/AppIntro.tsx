import { createMQ } from '@allthings/elements/Responsive'
import { Spacer, Text, View } from '@allthings/elements'
import { css } from 'glamor'
import React from 'react'
import { connect } from 'react-redux'

const styles = {
  subtitle: css({
    fontWeight: '300',
    lineHeight: '1.1em',
    fontSize: '24px !important',
    [createMQ('desktop', 'tablet')]: {
      fontSize: '34px !important',
    },
  }),
  title: css({
    fontWeight: '700',
    lineHeight: '1.1em',
    fontSize: '38px !important',
    hyphens: 'auto',
    [createMQ('desktop', 'tablet')]: {
      fontSize: '54px !important',
    },
  }),
}

type Props = ReturnType<typeof mapStateToProps>

class AppIntro extends React.PureComponent<Props> {
  public render(): JSX.Element {
    const { appName, appSubTitle, appTitle, language } = this.props

    return (
      <View direction="column">
        <Text
          data-e2e="login-intro-title"
          size="giant"
          lang={language}
          {...styles.title}
        >
          {appTitle || appName}
        </Text>
        <Spacer height="5vh" />
        <Text data-e2e="login-intro-subtitle" size="giant" {...styles.subtitle}>
          {appSubTitle}
        </Text>
      </View>
    )
  }
}

const mapStateToProps = (state: IReduxState) => ({
  appName: state.app.config.appName,
  appSubTitle:
    state.app.config.slogan[state.app.locale] ||
    state.app.config.slogan.en_US ||
    state.app.config.appSubTitle,
  appTitle: state.app.config.appTitle,
  language: (state.app.locale && state.app.locale.split('_')[0]) || 'en',
})

export default connect(mapStateToProps)(AppIntro)
