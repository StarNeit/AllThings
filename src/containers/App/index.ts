export { default as AppIcon } from './AppIcon'
export { default as AppIconButton } from './AppIconButton'
export { default as AppIntro } from './AppIntro'
export { default as AppTitle } from './AppTitle'
