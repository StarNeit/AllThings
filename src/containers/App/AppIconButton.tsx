// tslint:disable-next-line:no-implicit-dependencies
import { getIconSize, IProps as IconProps } from 'components/Icon'
import React from 'react'
import AppIcon from './AppIcon'

interface IProps {
  readonly background: string
  readonly size: IconProps['size']
  readonly style: object
}

class AppIconButton extends React.PureComponent<IProps> {
  public static readonly defaultProps: Partial<IProps> = {
    background: null,
  }

  public render(): JSX.Element {
    const iconSize = getIconSize(this.props.size)

    return (
      <AppIcon
        {...this.props}
        style={{
          ...this.props.style,
          background: this.props.background,
          height: iconSize,
          padding: Math.max((iconSize - 30) / 2, 0),
          width: iconSize,
        }}
      />
    )
  }
}

export default AppIconButton
