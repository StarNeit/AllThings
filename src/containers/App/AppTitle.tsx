import React from 'react'
import { getLocalized } from 'containers/Localized'
import { Helmet } from 'react-helmet'
import { connect } from 'react-redux'

interface IProps {
  children: object | string
  locale: string
}

class AppTitle extends React.PureComponent<IProps> {
  render() {
    return (
      <Helmet
        title={
          (typeof this.props.children === 'string'
            ? this.props.children
            : getLocalized(this.props.children, this.props.locale)) as string
        }
      />
    )
  }
}

export default connect((state: IReduxState) => ({ locale: state.app.locale }))(
  AppTitle,
)
