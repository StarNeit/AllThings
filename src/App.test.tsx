import React from 'react'
import App from './App'

describe('Check the App', () => {
  it('should check the app', () => {
    const wrapper = global.shallowIntl(<App />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should propagate the translations down by having a validity object', () => {
    const wrapper = global.shallowIntl(<App />)
    expect(wrapper).toMatchSnapshot()
  })
})
