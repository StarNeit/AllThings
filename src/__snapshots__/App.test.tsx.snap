// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the App should check the app 1`] = `
<ContextProvider
  value={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatDateToParts": [Function],
      "formatHTMLMessage": [Function],
      "formatList": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatNumberToParts": [Function],
      "formatPlural": [Function],
      "formatRelativeTime": [Function],
      "formatTime": [Function],
      "formatTimeToParts": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getListFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralRules": [Function],
        "getRelativeTimeFormat": [Function],
      },
      "locale": "en-US",
      "messages": Object {},
      "onError": [Function],
      "textComponent": "span",
      "timeZone": undefined,
    }
  }
>
  <App />
</ContextProvider>
`;

exports[`Check the App should propagate the translations down by having a validity object 1`] = `
<ContextProvider
  value={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatDateToParts": [Function],
      "formatHTMLMessage": [Function],
      "formatList": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatNumberToParts": [Function],
      "formatPlural": [Function],
      "formatRelativeTime": [Function],
      "formatTime": [Function],
      "formatTimeToParts": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getListFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralRules": [Function],
        "getRelativeTimeFormat": [Function],
      },
      "locale": "en-US",
      "messages": Object {},
      "onError": [Function],
      "textComponent": "span",
      "timeZone": undefined,
    }
  }
>
  <App />
</ContextProvider>
`;
